src/Sarif/Visitors/ReformattingVisitor.cs

```diff
@@ -43,23 +45,129 @@ public override Run VisitRun(Run node)
                 }
             }
 
+            if (node.Results != null)
+            {
+                var results = new List<Result>();
+
+                foreach (Result result in node.Results)
+                {
+                    results.Add(VisitResult(result));
+                }
+
+                node.Results = results;
+            }
+
             return node;
         }
 
-        internal FileData VisitDictionaryValueNullChecked(string key, FileData node)
+        public override PhysicalLocation VisitPhysicalLocation(PhysicalLocation node)
         {
-            Uri uri;
+            // If we have a binary region, there is no work to do
+            if (node.Region == null ||
+                (node.Region.ByteOffset > 0 || node.Region.ByteLength > 0))
+            {
+                goto Exit;
+            }
+
+            // If we already have a snippet, let's assume it is correct
+            if (node.Region.Snippet != null)
+            {
+                goto Exit;
+            }
 
-            if (!Uri.TryCreate(key, UriKind.RelativeOrAbsolute, out uri))
+            // If the user hasn't specified code snippet insertion, there is no work to do
+            bool workToDo = false;
+            workToDo |= node.Region != null && _dataToInsert.Includes(OptionallyEmittedData.CodeSnippets);
+
+            if (!workToDo)
```

Comment:
Or just:
```
if (node.Region == null ||  !_dataToInsert.Include(...))
```

Reply:
Right. This is following the workToDo pattern established elsewhere in this file. The thought is that this code will be subsequently updated to include multiline code snippets, so the logical will be built out further.

---
In reply to: [202101477](https://github.com/Microsoft/sarif-sdk/pull/942#discussion_r202101477) [](ancestors = 202101477)